mailbox/api/src/main/java/org/apache/james/mailbox/model/AttachmentId.java

```diff
@@ -20,20 +20,44 @@
 
 
 import java.nio.charset.StandardCharsets;
+import java.util.Optional;
 import java.util.UUID;
 
 import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.james.mime4j.codec.DecodeMonitor;
+import org.apache.james.mime4j.field.ContentTypeFieldImpl;
+import org.apache.james.mime4j.stream.RawField;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Objects;
 import com.google.common.base.Preconditions;
 import com.google.common.base.Strings;
+import com.google.common.primitives.Bytes;
 
 public class AttachmentId {
 
-    public static AttachmentId forPayload(byte[] payload) {
+    private static final String DEFAULT_MIME_TYPE = "text/html";
```

Comment:
why not text/plain? or binary/application-data?

Reply:
Ok, the generic type seems to be application/octet-stream